django/website/logframe/views.py

```diff
@@ -64,18 +65,30 @@ def get_data(self, logframe, data):
         return data
 
 
-class CreateLogframe(PermissionRequiredMixin, CreateView):
+class ManageLogFrame(PermissionRequiredMixin):
     model = LogFrame
-    form_class = CreateLogFrameForm
-    template_name = 'logframe/create_logframe.html'
     permission_required = 'logframe.edit_logframe'
 
     def get_success_url(self):
         return reverse('logframe-dashboard', kwargs={'slug': self.object.slug})
 
+class CreateLogframe(ManageLogFrame, CreateView):
+    form_class = CreateLogFrameForm
+    template_name = 'logframe/create_logframe.html'
+
+
     def form_valid(self, form):
         # The logframe needs to exist before we create the settings, so finish
         # the regular form_valid code.
         response = CreateView.form_valid(self, form)
         Settings.objects.create(logframe=form.instance)
         return response
+
+
+class EditLogframe(ManageLogFrame, UpdateView):
+    fields = ['name', 'slug']
+    template_name = 'logframe/edit_logframe.html'
+
+
+class DeleteLogframe(ManageLogFrame, DeleteView):
+    pass
```

Comment:
If I delete a log frame, I get redirected to the URL of the log frame I just deleted and a 404